source/h5vol/H5Vol_dataset.c

```diff
@@ -85,20 +85,30 @@ void *H5VL_adios2_dataset_open(void *obj, const H5VL_loc_params_t *loc_params,
     return result;
 }
 
-herr_t H5VL_adios2_dataset_read(void *dset, hid_t mem_type_id,
-                                hid_t mem_space_id, hid_t file_space_id,
-                                hid_t plist_id, void *buf, void **req)
+herr_t H5VL_adios2_dataset_read(size_t count, void *dset_array[],
+                                hid_t mem_type_id_array[],
+                                hid_t mem_space_id_array[],
+                                hid_t file_space_id_array[], hid_t dxpl_id,
+                                void *buf_array[],
+                                void **req) // last parameter is not unused
+                                            // void *dset, hid_t mem_type_id,
+// hid_t mem_space_id, hid_t file_space_id,
+// hid_t plist_id, void *buf, void **req)
 {
-    REQUIRE_NOT_NULL_ERR(dset, -1);
-    H5VL_ObjDef_t *vol = (H5VL_ObjDef_t *)dset;
+    for (size_t i = 0; i < count; i++)
+    {
+        REQUIRE_NOT_NULL_ERR(dset_array[i], -1);
+        H5VL_ObjDef_t *vol = (H5VL_ObjDef_t *)(dset_array[i]);
 
-    H5VL_VarDef_t *var = (H5VL_VarDef_t *)(vol->m_ObjPtr);
+        H5VL_VarDef_t *var = (H5VL_VarDef_t *)(vol->m_ObjPtr);
 
-    var->m_HyperSlabID = file_space_id;
-    var->m_MemSpaceID = mem_space_id;
+        var->m_HyperSlabID = file_space_id_array[i];
+        var->m_MemSpaceID = mem_space_id_array[i];
 
-    var->m_Data = buf;
-    return gADIOS2ReadVar(var);
+        var->m_Data = buf_array[i];
+        gADIOS2ReadVar(var);
```

Comment:
```suggestion
        herr_t ret = ADIOS2ReadVar(var);
        
        // A herr_t variable is considered successful only when > 0
        if (ret < 0)
        {
           return ret; 
        }
```

Reply:
A note for the future: according to hdf5 docs: https://docs.hdfgroup.org/hdf5/develop/_h5public_8h.html#title52 . A herr_t represents failure when <= 0, but gADIOS2ReadVar considers == 0 success. No something to change in this MR, but it will be good to address it in the future either getting rid of using herr_t in our codebase or following the hdf5 usage of it.

Reply:
Thanks for the suggestions. I removed the comments etc and adjusted return code. To clarify, the definition of success in hdf5 is 0 as defined in H5private.h.